uefi/src/proto/device_path/text.rs

```diff
@@ -16,32 +16,34 @@ use core::ops::Deref;
 use core::ptr::NonNull;
 use uefi_raw::protocol::device_path::{DevicePathFromTextProtocol, DevicePathToTextProtocol};
 
-/// This struct is a wrapper of `display_only` parameter
-/// used by Device Path to Text protocol.
+/// Parameter for [`DevicePathToText`] that alters the output format.
 ///
-/// The `display_only` parameter controls whether the longer
-/// (parseable)  or shorter (display-only) form of the conversion
-/// is used. If `display_only` is TRUE, then the shorter text
-/// representation of the display node is used, where applicable.
-/// If `display_only` is FALSE, then the longer text representation
-/// of the display node is used.
+/// * `DisplayOnly(false)` produces parseable output.
+/// * `DisplayOnly(true)` produces output that _may_ be shorter and not
+///   parseable.
+///
+/// Example of how a node's text representation may be altered by this
+/// parameter:
+/// * `DisplayOnly(false)`: `Ata(Primary,Master,0x1)`
+/// * `DisplayOnly(true)`: `Ata(0x1)`
 #[derive(Clone, Copy, Debug)]
 pub struct DisplayOnly(pub bool);
 
-/// This struct is a wrapper of `allow_shortcuts` parameter
-/// used by Device Path to Text protocol.
+/// Parameter for [`DevicePathToText`] that alters the output format.
+///
+/// * `AllowShortcuts(false)`: node names are based only on the node's type and
+///   subtype.
+/// * `AllowShortcuts(true)` _may_ alter the node name based on other fields
+///   within the node.
 ///
-/// The `allow_shortcuts` is FALSE, then the shortcut forms of
-/// text representation for a device node cannot be used. A
-/// shortcut form is one which uses information other than the
-/// type or subtype. If `allow_shortcuts is TRUE, then the
-/// shortcut forms of text representation for a device node
-/// can be used, where applicable.
+/// Example of how a node's text representation may be altered by this
+/// parameter:
+/// * `AllowShortcuts(false)`: `VenMsg(E0C14753-F9BE-11D2-9A0C-0090273FC14D)`
+/// * `AllowShortcuts(true)`: `VenPcAnsi()`
 #[derive(Clone, Copy, Debug)]
 pub struct AllowShortcuts(pub bool);
 
-/// Wrapper for a string internally allocated from
-/// UEFI boot services memory.
+/// UCS-2 string allocated from UEFI pool memory.
```

Comment:
General question: Why do we have this type? I think, this can also be a `CString16`?

Do we at least want to write how it differs from CString16?

Reply:
Updated the docstring to clarify. Since this is allocated internally by UEFI, we'd have to do a second memory allocation to copy it to a `CString16`. (Or alternatively, the `CString16` type could be modified to contain either a `Vec` or a pool allocation, and switch out how it's dropped at the end, but that's probably more complicated than it's worth.) I'm not necessarily opposed to doing something else here, like just accept an extra memory copy, but should probably think about that as part of https://github.com/rust-osdev/uefi-rs/issues/1303 since `CString16` requires the `alloc` feature.